framework/mongo/utils.py

```diff
@@ -63,10 +63,18 @@ def wrapper(cls):
 
 def get_or_http_error(Model, pk):
     instance = Model.load(pk)
+
     if getattr(instance, 'is_deleted', False):
-        raise HTTPError(http.GONE, data=dict(
-            message_long="This resource has been deleted"
-        ))
+        if hasattr(instance, 'registrations'):
+            raise HTTPError(http.GONE, data=dict(
+                message_short="Project Deleted",
+                message_long="This Project has been deleted"
```

Comment:
@sloria ... I think @lyndsysimon has made a good point on the original GH Issue #3421 re: disclosure of elements and how distinct we need to be the no longer existent "resource" message. Can you give an opinion on that? Maybe we get insight from others?